src/utils/timezone_utils.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+export function getUserCurrentTimezone(userTimezone) {
+    if (!userTimezone) {
+        return null;
+    }
+    const {
+        useAutomaticTimezone,
+        automaticTimezone,
+        manualTimezone,
+    } = userTimezone;
+
+    let useAutomatic = useAutomaticTimezone;
+    if (typeof useAutomaticTimezone === 'string') {
+        useAutomatic = useAutomaticTimezone === 'true';
+    }
+
+    if (useAutomatic) {
+        return automaticTimezone;
+    }
+    return manualTimezone;
+}
+
+export function getTimezoneRegion(timezone) {
+    if (timezone) {
+        const split = timezone.split('/');
+        if (split.length > 1) {
+            const region = split.pop();
+            if (region.indexOf('_') >= 0) {
```

Comment:
nit: is the check here worth it? Seems like an unconditional `replace` might just be clearer:
```
return split.pop().replace('_', ' ');
```